### DISCUSSION TITLE
Best practice for 100% bypass URIs

### DISCUSSION THREAD

**Discussion by 2q2code** [authelia/authelia#8416] [Q&A] [answered]
I use authelia to secure sites that do not have suitable or secure-enough authentication mechanisms of their own. On sites that do, I am a bit torn on how to proceed. I am currently doing both of these, but only really need to do one. For best practice should I:

A. Add the site into a bypass policy rule in my authelia configuration, or
B. Skip authelia's involvement in handling anything to do with the site (i.e. exclude it from the proxy configuration)

On one hand, I would prefer to handle all site access management in one place, but on the other hand, skipping the involvement of authelia altogether reduces the number of moving parts, and request overheads, for sites that are open to the public, or that rely on their own authentication mechanisms.

Is there an official guideline or stance on this?

Is there a community guideline or stance on this?

Thank you.

**Comment by Crowley723** [comment, marked as answer]
This is more of a personal answer rather than as a contributor.

I don't see any difference between having the Authelia middleware (configured with bypass) on a service vs not having the middleware at all. Authelia is so fast that unless I am running a **really** high volume web service it won't make a difference.

The benefit of having it in Authelia means you can be more fine grained then you can otherwise. Want to disable a specific API endpoint? you can do that with Authelia. Want to block access to a particular resource? You can do that too.

Your question sounds pretty philosophical, realistically I don't think you will see any difference between having the attached middleware or not.

**Reply by 2q2code** [reply]
Or perhaps if you're running on reduced resources like a Raspberry Pi 4 with 2GB of RAM. As it happens, there is a bug in my Authelia configuration at the moment and I've had to disable it completely. Frustrating, because I've been using Authelia for years without issue. I rebuilt the configuration from the ground up for this new system build and ... but that's a topic for another discussion. Thanks.

**Comment by hendrik1120** [comment]
Fully agree with @Crowley723. 
Because you mentioned reducing the number of moving parts, it does make a difference there.
In the case of authelia being unavailable, you want to consider what happens to your application. Which depends on how reliant your application is on authelia. If it is just using authelia for OpenID Connect, you might not want the whole application to go down just because one identity provider is not available.
On the other hand, you might want your application to be down, because there is no alternative login method and it would be unusable without.

I would say that it depends on your desired failure behavior to either fail closed or open.
